plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedViews.java

```diff
@@ -140,6 +140,15 @@ public void testCreateWithInvalidPropertyFails()
                 .hasMessage("Catalog 'iceberg' does not support materialized view property 'invalid_property'");
     }
 
+    @Test
+    public void testCreateWithDuplicateSourceTableSucceeds()
+    {
+        assertUpdate("CREATE MATERIALIZED VIEW materialized_view_with_duplicate_source AS " +
```

Comment:
The changed code isn't going to get called unless we also do a `assertUpdate("REFRESH MATERIALIZED VIEW materialized_view_with_duplicate_source", 12);` after the create.
But the test doesn't fail even after adding that without your fix.
We need a test which would fail without your fix.

Reply:
ok I see, I need to refresh and then select from the table. This then should fail as the metadata is only evaluated at query time not when refreshing.

Reply:
select path added which should trigger both the write metadata and read metadata path.